examples/finance/bs_ivbp.ipynb

```diff
@@ -0,0 +1,649 @@
+     "text": [
+      "dt,tmax,nt; 0.0005 1.0 2001\n",
+      "shape;    (81,)\n",
+      "origin;   (60,)\n",
+      "spacing;  (1.0,)\n",
+      "extent;   80\n"
+     ]
+    }
+   ],
+   "source": [
+    "from devito import (Eq, Grid, TimeFunction, Operator, solve, Constant, \n",
+    "                    SpaceDimension, configuration, SubDomain)\n",
+    "\n",
+    "from mpl_toolkits.mplot3d import Axes3D\n",
+    "from mpl_toolkits.mplot3d.axis3d import Axis\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib as mpl\n",
+    "from matplotlib import cm\n",
+    "\n",
+    "from sympy.stats import Normal, cdf\n",
+    "import numpy as np\n",
+    "import time as timer\n",
+    "\n",
+    "%matplotlib inline\n",
+    "mpl.rc('font', size=14)\n",
+    "\n",
+    "configuration[\"log-level\"] = 'INFO'\n",
+    "\n",
+    "# Constants\n",
+    "K = 100.0\n",
+    "r = 0.12\n",
+    "sigma = 0.1\n",
+    "smin = 70\n",
+    "smax = 130\n",
+    "\n",
+    "# If you want to try some different problems, uncomment these lines\n",
+    "\n",
+    "## Example 2\n",
+    "# K = 10.0\n",
+    "# r = 0.1\n",
+    "# sigma = 0.2\n",
+    "# smin = 0.0\n",
+    "# smax = 20.0\n",
+    "\n",
+    "## Example 3\n",
+    "# K = 100.0\n",
+    "# r = 0.05\n",
+    "# sigma = 0.25\n",
+    "# smin = 50.0\n",
+    "# smax = 150.0\n",
+    "\n",
+    "# Amount of padding to proccess left/right (hidden from graphs)\n",
+    "padding = 10\n",
+    "smin -= padding\n",
+    "smax += padding\n",
+    "\n",
+    "# Extent calculations\n",
+    "tmax = 1.0\n",
+    "dt0 = 0.0005\n",
+    "ds0  = 1.0\n",
+    "nt = (int)(tmax / dt0) + 1\n",
+    "ns = int((smax - smin) / ds0) + 1 \n",
+    "\n",
+    "shape = (ns, )\n",
+    "origin =(smin, )\n",
+    "spacing = (ds0, )\n",
+    "extent = int(ds0 * (ns - 1))\n",
+    "\n",
+    "print(\"dt,tmax,nt;\", dt0,tmax,nt)\n",
+    "print(\"shape;   \", shape)\n",
+    "print(\"origin;  \", origin)\n",
+    "print(\"spacing; \", spacing)\n",
+    "print(\"extent;  \", extent)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Black Scholes initial boundary value problem\n",
+    "$$\n",
+    "- \\frac{\\partial}{\\partial t} V(t,s) + r\\ s\\ \\frac{\\partial}{\\partial s} V(t,s) + \\frac{1}{2}\\ \n",
+    "    \\sigma^2\\ s^2\\ \\frac{\\partial^2}{\\partial s^2} V(t,s) - r\\ V(t,s) = 0\n",
+    "$$\n",
+    "\n",
+    "This differs from the regular Black Scholes PDE since our $\\frac{\\partial}{\\partial t} V(t,s)$ is negative. This is because instead of using a terminal condition, we are using a initial condition and solving backward in time.\n",
+    "\n",
+    "#### Update equation\n",
+    "\n",
+    "In our update equation, we solve for $v.\\text{forward} = v{\\left(t + dt,s \\right)}$.\n",
+    "\n",
+    "Because $$-\\frac{\\partial}{\\partial t} V(t,s) = \\displaystyle \\frac{v{\\left(t,s \\right)}}{dt} - \\frac{v{\\left(t + dt,s \\right)}}{dt}$$\n",
```

Comment:
Note the order of FD approximations.

Reply:
Noted in the equation section and boundary condition section